### DISCUSSION TITLE
GoAccess Feed by DB instead of logs

### DISCUSSION THREAD

**Discussion by openHBP** [allinurl/goaccess#2237] [Q&A] [answered]
Is it possible to run GoAccess with his own database files (I32_DATES.db, IGLP_LAST_PARSE.db, ect.) as input instead of the log files?

I've followed [Arnaud R](https://arnaudr.io/2020/08/10/goaccess-14-a-detailed-tutorial/) detail tutorial and it works fine.

I've been able to generate **monthly report** using shell script but I also would like to keep a **yearly report.**

With an active logrotate on the log files we are loosing the history in the log folder (/var/log/nginx/site1)

However with the use of _--persit option_, GoAccess DB (/var/lib/goaccess-db/site1) keep the history. Before starting to keep my log on a yearly basis, I'm wondering whether it would be possible to run GoAccess against his own generated DB instead of the logs???

=> Run GoAccess on the aggregated monthly DB to get a yearly report

or

=> run GoAccess on the yearly DB with generation of a monthly output???

Thanks for this marvelous tools and for your insights

**Comment by allinurl** [comment, marked as answer]
Yes, you should be able to simply run goaccess against its own db files without any logs. e.g.,

    goaccess --restore --db-path=<dir> --log-format=COMBINED -o report.html

Give it a shot and let me know if that's what you are after. Thanks

**Reply by openHBP** [reply]
Excellent it works perfect! Thanks
